### ISSUE TITLE
Random Fatal Errors while testing

### ISSUE THREAD

**Issue by Mai-Saad** [wp-media/wp-rocket#8870]
- During manual or automated e2e tests, some random errors occur.
- Goal : list those errors and check if we need/can guard
1- BerlinDB related, https://group-onecom.slack.com/archives/C08F4LZB2QG/p1789056244612789?thread_ts=1748594719.779579&cid=C08F4LZB2QG
```
[17-Sep-2026 08:08:37 UTC] PHP Fatal error:  Uncaught Error: Class "WP_Rocket\Dependencies\BerlinDB\Database\Queries\Date" not found in /var/www/rucss.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Dependencies/BerlinDB/Database/Query.php:718
Stack trace:
#0 /var/www/rucss.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Dependencies/BerlinDB/Database/Query.php(1383): WP_Rocket\Dependencies\BerlinDB\Database\Query->get_date_query()
#1 /var/www/rucss.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Dependencies/BerlinDB/Database/Query.php(954): WP_Rocket\Dependencies\BerlinDB\Database\Query->parse_where()
#2 /var/www/rucss.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Dependencies/BerlinDB/Database/Query.php(902): WP_Rocket\Dependencies\BerlinDB\Database\Query->get_item_ids()
#3 /var/www/rucss.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Dependencies/BerlinDB/Database/Query.php(360): WP_Rocket\Dependencies\BerlinDB\Database\Query->get_items()
#4 /var/www/rucss.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Engine/Common/Database/Queries/AbstractQuery.php(32): WP_Rocket\Dependencies\BerlinDB\Database\Query->query()
#5 /var/www/rucss.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Engine/Preload/Database/Queries/Cache.php(720): WP_Rocket\Engine\Common\Database\Queries\AbstractQuery->query()
#6 /var/www/rucss.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Engine/Preload/Controller/PreloadUrl.php(261): WP_Rocket\Engine\Preload\Database\Queries\Cache->get_outdated_in_progress_jobs()
#7 /var/www/rucss.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Engine/Preload/Cron/Subscriber.php(102): WP_Rocket\Engine\Preload\Controller\PreloadUrl->process_pending_jobs()
#8 /var/www/rucss.rocketlabsqa.ovh/htdocs/wp-includes/class-wp-hook.php(353): WP_Rocket\Engine\Preload\Cron\Subscriber->process_pending_urls()
#9 /var/www/rucss.rocketlabsqa.ovh/htdocs/wp-includes/class-wp-hook.php(377): WP_Hook->apply_filters()
#10 /var/www/rucss.rocketlabsqa.ovh/htdocs/wp-includes/plugin.php(571): WP_Hook->do_action()
#11 /var/www/rucss.rocketlabsqa.ovh/htdocs/wp-cron.php(191): do_action_ref_array()
#12 {main}
  thrown in /var/www/rucss.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Dependencies/BerlinDB/Database/Query.php on line 718
```

2- Action scheduler related {when replace uploaded version} , similar to https://group-onecom.slack.com/archives/C08EFCYRQ2X/p1775043510731639?thread_ts=1775042713.115059&cid=C08EFCYRQ2X
```
[14-Sep-2026 09:42:40 UTC] PHP Fatal error:  Uncaught Error: Class "ActionScheduler_StoreSchema" not found in /var/www/new.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Dependencies/ActionScheduler/classes/data-stores/ActionScheduler_DBStore.php:63
Stack trace:
#0 /var/www/new.rocketlabsqa.ovh/htdocs/wp-includes/class-wp-hook.php(351): ActionScheduler_DBStore->init()
#1 /var/www/new.rocketlabsqa.ovh/htdocs/wp-includes/class-wp-hook.php(377): WP_Hook->apply_filters()
#2 /var/www/new.rocketlabsqa.ovh/htdocs/wp-includes/plugin.php(523): WP_Hook->do_action()
#3 /var/www/new.rocketlabsqa.ovh/htdocs/wp-settings.php(779): do_action()
#4 /var/www/new.rocketlabsqa.ovh/htdocs/wp-config.php(335): require_once('...')
#5 /var/www/new.rocketlabsqa.ovh/htdocs/wp-load.php(50): require_once('...')
#6 /var/www/new.rocketlabsqa.ovh/htdocs/wp-admin/admin-ajax.php(22): require_once('...')
#7 {main}
  thrown in /var/www/new.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Dependencies/ActionScheduler/classes/data-stores/ActionScheduler_DBStore.php on line 63
```

And this when delete plugin, similar to https://group-onecom.slack.com/archives/C08EFCYRQ2X/p1775057296429349?thread_ts=1775042713.115059&cid=C08EFCYRQ2X

```
[14-Sep-2026 12:31:27 UTC] PHP Fatal error:  Uncaught Error: Class "ActionScheduler_Lock" not found in /var/www/new.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Dependencies/ActionScheduler/classes/abstracts/ActionScheduler.php:55
Stack trace:
#0 /var/www/new.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Dependencies/ActionScheduler/classes/ActionScheduler_QueueRunner.php(142): ActionScheduler::lock()
#1 /var/www/new.rocketlabsqa.ovh/htdocs/wp-includes/class-wp-hook.php(353): ActionScheduler_QueueRunner->maybe_dispatch_async_request()
#2 /var/www/new.rocketlabsqa.ovh/htdocs/wp-includes/class-wp-hook.php(377): WP_Hook->apply_filters()
#3 /var/www/new.rocketlabsqa.ovh/htdocs/wp-includes/plugin.php(523): WP_Hook->do_action()
#4 /var/www/new.rocketlabsqa.ovh/htdocs/wp-includes/load.php(1308): do_action()
#5 [internal function]: shutdown_action_hook()
#6 {main}
  thrown in /var/www/new.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Dependencies/ActionScheduler/classes/abstracts/ActionScheduler.php on line 55
```

3- other
```
[14-Sep-2026 07:18:27 UTC] PHP Fatal error:  Uncaught Error: Class "WP_Rocket\Engine\Container\IndexedDefinitionAggregate" not found in /var/www/new.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/main.php:69
Stack trace:
#0 /var/www/new.rocketlabsqa.ovh/htdocs/wp-includes/class-wp-hook.php(353): rocket_init()
#1 /var/www/new.rocketlabsqa.ovh/htdocs/wp-includes/class-wp-hook.php(377): WP_Hook->apply_filters()
#2 /var/www/new.rocketlabsqa.ovh/htdocs/wp-includes/plugin.php(523): WP_Hook->do_action()
#3 /var/www/new.rocketlabsqa.ovh/htdocs/wp-settings.php(630): do_action()
#4 /var/www/new.rocketlabsqa.ovh/htdocs/wp-config.php(335): require_once('...')
#5 /var/www/new.rocketlabsqa.ovh/htdocs/wp-load.php(50): require_once('...')
#6 /var/www/new.rocketlabsqa.ovh/htdocs/wp-login.php(12): require('...')
#7 {main}
  thrown in /var/www/new.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/main.php on line 69
```

**Comment by DahmaniAdame** [comment]
@hellofromahmed do we need to guard against this? as it's only reported from QA for now.
